setuptools/command/build_py.py

```diff
@@ -50,20 +49,20 @@ def finalize_options(self):
             del self.__dict__['data_files']
         self.__updated_files = []
 
-    def copy_file(  # type: ignore[override] # No overload, str support only
+    def copy_file(  # type: ignore[override] # No overload, no bytes support
         self,
         infile: StrPath,
-        outfile: StrPath,
+        outfile: StrPathT,
         preserve_mode: bool = True,
         preserve_times: bool = True,
         link: str | None = None,
         level: object = 1,
-    ):
+    ) -> tuple[StrPathT | str, bool]:
         # Overwrite base class to allow using links
         if link:
             infile = str(Path(infile).resolve())
-            outfile = str(Path(outfile).resolve())
-        return super().copy_file(
+            outfile = str(Path(outfile).resolve())  # type: ignore[assignment] # Re-assigning a str when outfile is StrPath is ok
```

Comment:
This may or may not also be related to https://github.com/python/mypy/issues/18075